### DISCUSSION TITLE
Using Terrafarm explosive aka TerraBoom

### DISCUSSION THREAD

**Discussion by JohannesSpg** [scfmod/FS22_TerraFarm#42] [Ideas]
New to Github, so i hope i placed it in the right spot.....
I am thinking a few days about the Terrafarm mod now.
And i wonder if it can be transformed into a kind of explosive, where you dig holes in game.
Fill these holes with ANFO, Ammonium Nitrate and Fuel Oil.
Then attach a detonator to the hole(s), in effect the terrafarming device....
Blow it up and at the moment of detonation the terrafarming takes place.
Yes it requires some mods to develop like drilling rigs, ANFO truck and most of all the detonators, but it prevents that every digger have to be added individually to the compatibilty list or whatever, it can simplify things a lot.
Diesel (Fuel Oil) and Fertilizer (Ammonium Nitrate) we already have in game, just need to develop 3 mods basically and the terrafarm mod itself.
After the blow up, you can dig up the converted terrafarmed dirt, stone or whatever.
It comes closest to real life mining and can be used to terrafarm sections of land big or small.
1 hole or 100 holes depending on the system FS runs on and how fast it is.
Just an idea.

**Comment by scfmod** [comment]
It's definitely doable to make, but would make more sense to have it as a separate mod :-)
You only need to calculate your destruction radius/shape, and alter the density map for terrain height accordingly plus the density map for the fill type.
g_densityMapHeightManager should have most of the values and functions you need

**Comment by JohannesSpg** [comment]
Thinking to what we can tie the terrafarming script, can it be made like this that the blast radius is for example 5 or 10 meters, perhaps both?
And is it possible to have the script on a detonator so we can create is as a separate mod as suggested?
Because i am thinking of using this as a batch of 1 piece to a 100 or even more, each with a blast/terrafarming radius of 5 or 10 meters, this means for instance when you use 100 of them an area of 500 or 1000 square meters and a depth of 5 or 10 meters would release a maximum of 10.000 cubic meters if i am correct. Maybe it is too much lol.

**Comment by Dzi4d3k** [comment]
@JohannesSpg 
Please don't get me wrong, but modding doesn't work in such way as you doing right now. You simply can't come to a skilled modder and ask him or even propose to do any mods.
Modding is a pice of passion and it is done in modder's free time and obviously on what he willing to work on.
That is general idea, of course modders are often open for suggestions, but that's it.

**Reply by JohannesSpg** [reply]
Working on mods now with a couple of modders, one is good in mobile mods, other is good in placeables.
I respect them and treat them normally.
I can perfectly request some things, as i provide them with models to base the mods/placeables on.
With a modeler building from scratch i have no experience yet, but that can change.
Thanks for the feedback sir.

**Comment by scfmod** [comment]
The issue with converting blast radius (m3) to a fill type, i.e stone, is that the ground density has a height limit.
So there's only x amount of liters of stone you can have per m2 on the ground, depending on the map.

**Comment by Dzi4d3k** [comment]
@JohannesSpg No I don't mind that this is bad idea, in fact I think it's good idea. And from what I see @scfmod is more than helpful on your idea. Believe me or not, but with many other modders you would meet with rough attitude to such propositions.
As you wrote, that you are new to GitHub, so it's not bad to ask and discus about mods and enhancements, but it's not welcome to see demands.
I believe that is not excluded that @scfmod (of course I don't want to speak on his behalf) could consider you ideas as future enhancements, however at current state there are lot of base functionalities at the top of priority list :)

**Comment by JohannesSpg** [comment]
@Dzi4d3k i am new here, but not entirely new to the modding world and modders, some have a bad attitude, but most do not luckily, but if i step on long toes that is not my problem to be honest.
I have a few very skillful modders on my side who can help develop this idea further.
But you are right that it is maybe too soon to do that, the priorities are different at this moment.
For that reason i closed the topic already and let scfmod do his thing for now.
I am busy myself with bringing certain mods ingame and later we will see when the time is right.

**Reply by JohannesSpg** [reply]
Do we know where scfmod has gone?
He appears not active since January.

**Comment by scfmod** [comment]
No toes got mutilated 😅 I won't be adding such a feature to TerraFarm as it's more focused on machines :-)
But the idea is interesting, and totally feasible to code. Bascially everything you'd need to make this work is in the DensityMapHeightManager class. DensityMapHeightUtil also contains a lot of helper functions.

**Comment by scfmod** [comment]
Closing this issue.
I added the Discussions feature to the repository so that you can use that instead of Issues section :-)

**Comment by JohannesSpg** [comment]
scfmod can you adjust the code to add a trigger action for the terrafarming to take place?
Not using a bucket but a detonator, for that i need a trigger input.
Can that be made?
I hope you are still active.

**Reply by JohannesSpg** [reply]
Cleared out messages that are not relevant to the topic.

**Reply by JohannesSpg** [reply]
The point is, i not use a bucket or a ripper, that you push into the ground with a machine.
With the machine you perform the action, the terrafarm is tied to the bucket/ripper.
As soon as it hits the soil the terrafarm action takes place.
I want terrafarm to take place when the trigger is placed underground and remains inactive until the trigger is pulled so to speak.
That is why i need a triggerinput.
